src/com/selftest/trafficlight/TrafficLightSimulator.java

```diff
@@ -0,0 +1,62 @@
+package com.selftest.trafficlight;
+
+public class TrafficLightSimulator implements Runnable {
+
+	private Thread t;
+	private TrafficLightColour tlc;
+	boolean stop = false;
+	boolean changed = false;
+
+	public TrafficLightSimulator(TrafficLightColour init) {
+		tlc = init;
+
+		t = new Thread(this);
+		t.start();
+	}
+
+	public TrafficLightSimulator() {
+		this(TrafficLightColour.RED); // default to RED
+	}
+
+	@Override
+	public void run() {
+		while (!stop) {
+			try {
+				Thread.sleep(tlc.getSleepMillis());
+			} catch (InterruptedException e) {
+				System.out.println(e);
+			}
+			changeColour();
+		}
+		
+	}
+
+	public synchronized void changeColour() {
+		// Excuses to use ordinals etc
+		tlc = (tlc.ordinal() == 2) ? TrafficLightColour.values()[0]

```

Comment:
But what happens when Oracle realises their's more to the world than just U.S.A.? And are forced kicking and screaming to add the colour "RED_n_AMBER"? 
You'll have to change this code.